packages/components/data-table/src/column-size-utils.js

```diff
@@ -0,0 +1,35 @@
+const MINIMUM_COLUMN_SIZE = 24;
+
+// calculates size on mouse-drag and enforces a minimum size
+const calculateResize = (
+  initialSize,
+  initialMousePosition,
+  newMousePosition,
+  minSize = MINIMUM_COLUMN_SIZE
+) => {
+  const newSize = initialSize - (initialMousePosition - newMousePosition);
+
+  return minSize > newSize ? minSize : newSize;
+};
+
+// returns an array of column widths
+const getColumnWidthsArray = (columns) =>
+  columns.reduce((widths, current) => [...widths, current.width], []);
+
+const setColumnWidth = (columns, position, value) => {
+  // eslint-disable-next-line no-param-reassign
+  columns[position] = value;
```

Comment:
I think this is one of those cases where mutating the value directly is advantageous because it's for adjusting the layout quickly on-the-fly, before any re-renders or reconciliations.